Signum.Engine/Operations/OperationLogic.cs

```diff
@@ -277,8 +277,25 @@ static void EntityEventsGlobal_Saving(Entity ident)
     #region Events
 
     public static event SurroundOperationHandler? SurroundOperation;
+    public static event OperationHandlerArgs? OperationBeforeExecute;

```

Comment:
Why not use SurroundOperationHandler?

Check DiffLocLogic OperationLogic_SurroundOperation. 

This this method you can execute stuff before (normal) after (using new Disposable(()=>{}))  and pass context from one method to the other.

Reply:
because SurroundOperationHandler is inside the transaction. While the new ones are out of the transaction